src/descriptor/key.rs

```diff
@@ -525,6 +525,53 @@ impl error::Error for ConversionError {
     }
 }
 
+pub trait DescriptorInnerKey {
```

Comment:
In e4d9af484b996d05db261be7406d2af78488e527:

This can go in a separate PR, but like to have a type-level relationship between `DescriptorPublicKey` and `DescriptorSecretKey`. This would help me downstream in elements-miniscript where I want to define a "view key" for a confidential transaction descriptor, which has a private key in one slot and a public key in every other slot. I think it would also simplify a lot of usage of xprv descriptors, which currently require you maintain an xpub descriptor alongside a hashmap, though I haven't worked out the details.

One way we could do this is to add two associated types, like
```
type SecretKey: DescriptorInnerKey;
type PublicKey: DescriptorInnerKey;
```
where almost always one of these would be `Self`.

I would also like first-class support for unspendable keys in taproot. I am imagining an `UnspendableKey` whose `SecretKey` type would be `!`, or something.

Anyway these are just half-baked ideas that don't belong in this PR. But I want to bring them up in case they could guide this API.